### PR TITLE
Problem: upgrade to rsa 0.5.0 fails(#143)

### PR DESCRIPTION
**Pull request by linfeng-crypto**

### CODE REVIEW

**Review by tomtau — CHANGES_REQUESTED** [review]
build error -- https://github.com/crypto-com/tmkms-light/pull/144/checks?check_run_id=3247940363#step:5:529
as rsa is used on sgx crates, you can try building those when testing your patch

File: providers/sgx/sgx-app/src/sgx_app/cloud.rs [hunk 1]
```diff
@@ -55,18 +56,12 @@ pub fn generate_keypair(
...
-                Ok((pub_key, sealed_secret, report))
-            }
-            Err(e) => {
-                secret.zeroize();
-                Err(CloudError::SealingError(e))
-            }
+    if let Ok(secret) = pkcs1 {
+        match crate::sgx_app::keypair_seal::seal_secret(csprng, secret.as_der(), result.into()) {
+            Ok(sealed_secret) => Ok((pub_key, sealed_secret, report)),
+            Err(e) => Err(CloudError::SealingError(e)),
```
Comment (tomtau) [inline providers/sgx/sgx-app/src/sgx_app/cloud.rs:64]: does the `secret` have zeroize on drop?

**Review by tomtau — COMMENTED** [review]
did you also try running the SGX tests (`cargo test --target x86_64-fortanix-unknown-sgx` in the sgx-app)? it needs to be on the machine with SGX support -- I can try to checkout and see if they still pass

File: providers/sgx/sgx-app/src/sgx_app/cloud.rs (hunk 1 above)
Comment (linfeng-crypto) [reply providers/sgx/sgx-app/src/sgx_app/cloud.rs:64]: yes, it's a wrapper of `Zeroizing<Vec<u8>>`

**Comment by linfeng-crypto** [comment]
> did you also try running the SGX tests (`cargo test --target x86_64-fortanix-unknown-sgx` in the sgx-app)? it needs to be on the machine with SGX support -- I can try to checkout and see if they still pass

no, I'll do it.

**Review by tomtau — APPROVED** [review]
just tried running the sgx tests:

**Pull request merged**